File: app/src/main/java/com/umbrella/budgetapp/Database/Collections/Category.java
package com.umbrella.budgetapp.Database.Collections;

public class Category {
    public static final String COLLECTION = "category";
    public static final String NAME = "name";
    public static final String COLOR = "color";
    public static final String ICON = "icon";

    private String Name;
    private int Icon;
    private int Color;

    public Category() {
        //Empty constructor needed.
    }

    public Category(String name, int icon, int color) {
        Name = name;
        Icon = icon;
        Color = color;
    }

    public String getName() {
        return Name;
    }

    public int getIcon() {
        return Icon;
    }

    public int getColor() {
        return Color;
    }

    public void setName(String name) {
        Name = name;
    }

    public void setIcon(int icon) {
        Icon = icon;
    }

    public void setColor(int color) {
        Color = color;
    }
}

File: app/src/main/java/com/umbrella/budgetapp/database/collections/Category.kt
package com.umbrella.budgetapp.database.collections

import android.os.Parcelable
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.android.parcel.Parcelize

@Parcelize
@Entity(tableName = "categories")
data class Category(
        @PrimaryKey(autoGenerate = true)
        @ColumnInfo(name = "category_id")
        val id: Long? = null,

        @ColumnInfo(name = "category_name")
        var name: String? = "",

        @ColumnInfo(name = "category_icon")
        var icon: Int? = 0,

        @ColumnInfo(name = "category_color")
        var color: Int? = 0
) : Parcelable